src/System.Resources.Reader/src/System/Resources/StreamExtensions.cs

```diff
@@ -78,13 +80,35 @@ public static string ReadString(this Stream stream)
                 int justRead = stream.Read(buffer, totalRead, stringLength - totalRead);
                 if(justRead == 0)
                 {
-                    throw new BadImageFormatException(SR.Resources_StreamNotValid);
+                    throw new EndOfStreamException(SR.BadImageFormat_ResourceNameCorrupted_NameIndex + position);
```

Comment:
I filed #4614 to think what to do here. In general, I will have to redo how resources are handled in resource.reader if we want to use resource.reader as the low level resource reading API.